netjsonconfig/backends/openwrt/converters/firewall.py

```diff
@@ -0,0 +1,70 @@
+from collections import OrderedDict
+
+from .base import OpenWrtConverter
+
+
+class Firewall(OpenWrtConverter):
+    netjson_key = 'firewall'
+    intermediate_key = 'firewall'
+    _uci_types = ['defaults', 'forwarding', 'zone']
+
+    def to_intermediate_loop(self, block, result, index=None):
+        forwardings = self.__intermediate_forwardings(block.pop('forwardings', {}))
+        zones = self.__intermediate_zones(block.pop('zones', {}))
+        block.update({
+            '.type': 'defaults',
+            '.name': block.pop('id', 'defaults'),
+        })
+        result.setdefault('firewall', [])
+        result['firewall'] = [self.sorted_dict(block)] + forwardings + zones
+        return result
+
+    def __intermediate_forwardings(self, forwardings):
+        """
+        converts NetJSON forwarding to
+        UCI intermediate data structure
+        """
+        result = []
+        for forwarding in forwardings:
+            resultdict = OrderedDict((('.name', self.__get_auto_name_forwarding(forwarding)),
+                                      ('.type', 'forwarding')))
+            for key, value in forwarding.items():
+                resultdict[key] = value
+            result.append(resultdict)
+        return result
+
+    def __get_auto_name_forwarding(self, forwarding):
+        if 'family' in forwarding.keys():
+            uci_name = self._get_uci_name('_'.join([forwarding['src'], forwarding['dest'],
+                                          forwarding['family']]))
+        else:
+            uci_name = self._get_uci_name('_'.join([forwarding['src'], forwarding['dest']]))
+        return 'forwarding_{0}'.format(uci_name)
+
+    def __intermediate_zones(self, zones):
+        """
+        converts NetJSON zone to
+        UCI intermediate data structure
+        """
+        result = []
+        for zone in zones:
+            resultdict = OrderedDict((('.name', self.__get_auto_name_zone(zone)),
+                                      ('.type', 'zone')))
+            for key, value in zone.items():
+                resultdict[key] = value
```

Comment:
Couldn't this be written as:
```python
resultdict.update(zone)
```

Reply:
Thank you @edoput and @AnonGuy for your reviews, I will look into this tomorrow.